.github/workflows/build_and_test.yml

```diff
@@ -89,6 +89,14 @@ jobs:
       # In order to fetch changed files
       with:
         fetch-depth: 0
+    - uses: actions/checkout@v2
+      name: checkout n1hility/cancel-previous-runs
+      with:
+        repository: n1hility/cancel-previous-runs
+        path: ./build/.actions/n1hility/cancel-previous-runs
+        ref: 953c92201f368370112ea2754545cb4468d89f12 # v2
+        fetch-depth: 1
+    - uses: ./build/.actions/n1hility/cancel-previous-runs
```

Comment:
Token is needed here: https://github.com/apache/spark/pull/31104/checks?check_run_id=1673964946#step:4:15

See how the token was declared: https://github.com/apache/spark/pull/31098/files#diff-48c0ee97c53013d18d6bbae44648f7fab9af2e0bf5b0dc1ca761e18ec5c478f2R94

Could you please add it to see if the action would be able to cancel your previous runs for this PR?

Reply:
This token is not needed. ;-) Each action gets a token by default in a variable named "Token". The action/checkout works the same way. See: https://github.com/actions/checkout/blob/c952173edf28a2bd22e1a4926590c1ac39630461/src/input-helper.ts#L109

Reply:
Token is required for cancel-previous-runs: https://github.com/n1hility/cancel-previous-runs/blob/178b93a12fb2731212e48c2cc5b7e37b937fd339/action.yml#L5-L7

Reply:
This value is passed to all actions by default. See for another example: https://github.com/JamesIves/github-pages-deploy-action/pull/530

Reply:
@mik-laj, please double-check the link I highlight: https://github.com/apache/spark/pull/31104/checks?check_run_id=1673964946#step:4:15

It literally says that `n1hility/cancel-previous-runs` fails and it requires authentication:

```
Run ./build/.actions/n1hility/cancel-previous-runs
  with:
  env:
    MODULES_TO_TEST: core, unsafe, kvstore, avro, network-common, network-shuffle, repl, launcher, examples, sketch, graphx
    EXCLUDED_TAGS: 
    INCLUDED_TAGS: 
    HADOOP_PROFILE: hadoop3.2
    HIVE_PROFILE: hive2.3
    CONDA_PREFIX: /usr/share/miniconda
    GITHUB_PREV_SHA: 50d17ba02600601f8f63fcd55c2bf70865f9dc6e
    GITHUB_INPUT_BRANCH: 
[@octokit/rest] `const Octokit = require("@octokit/rest")` is deprecated. Use `const { Octokit } = require("@octokit/rest")` instead

Branch is patch-1, repo is spark, and owner is apache, and id is 474259497
(node:1987) UnhandledPromiseRejectionWarning: Error: Parameter token or opts.auth is required
    at Function.getAuthString (/home/runner/work/spark/spark/build/.actions/n1hility/cancel-previous-runs/dist/index.js:5899:19)
    at Function.getOctokitOptions (/home/runner/work/spark/spark/build/.actions/n1hility/cancel-previous-runs/dist/index.js:5864:29)
    at new GitHub (/home/runner/work/spark/spark/build/.actions/n1hility/cancel-previous-runs/dist/index.js:5848:22)
    at /home/runner/work/spark/spark/build/.actions/n1hility/cancel-previous-runs/dist/index.js:1492:25
    at Generator.next (<anonymous>)
    at /home/runner/work/spark/spark/build/.actions/n1hility/cancel-previous-runs/dist/index.js:1448:71
    at new Promise (<anonymous>)
    at module.exports.131.__awaiter (/home/runner/work/spark/spark/build/.actions/n1hility/cancel-previous-runs/dist/index.js:1444:12)
    at cancelDuplicates (/home/runner/work/spark/spark/build/.actions/n1hility/cancel-previous-runs/dist/index.js:1491:12)
    at /home/runner/work/spark/spark/build/.actions/n1hility/cancel-previous-runs/dist/index.js:1600:13
(node:1987) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:1987) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

Apparently, that does not look like a workable action.

Reply:
I agree. I checked carefully and it looks like this action must be run as workflow_run.

> If you use forks, you should create a separate "Cancelling" workflow_run triggered workflow. The workflow_run should be responsible for all canceling actions. The examples below show the possible ways the action can be utilized.

https://github.com/potiuk/cancel-workflow-runs#usage

I am rewriting my PR now to use workflow_run. I will also ask @potiuk for a review when I'm finished.

Reply:
Yes. It must be a workflow_run action. Ant the 'cancel-workflow-run' action of mine has all the support for that.

Reply:
The n1hility action I based my action initially on should be deprecated as it has no good support for that. I actually created a ticket to either depreacate it it merge my changes: https://github.com/n1hility/cancel-previous-runs/issues/7 (I am happy my action to disapear and merge it back) but I have not heard from the author despite nagging several times

Reply:
I pushed a new change.s  I passed token and used workflow_run event.